src/components/calcite-inline-editable/calcite-inline-editable.tsx

```diff
@@ -0,0 +1,265 @@
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+
+  @Element() el!: HTMLCalciteInlineEditableElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Props
+  //
+  //--------------------------------------------------------------------------
+
+  /** specify whether the wrapped input element is editable, defaults to false */
+  @Prop({ mutable: true, reflect: true }) editingEnabled = false;
+
+  /** specify whether the confirm button should display a loading state, defaults to false */
+  @Prop({ mutable: true, reflect: true }) loading = false;
+
+  /** specify whether save/cancel controls should be displayed when editingEnabled is true, defaults to false */
+  @Prop({ reflect: true }) hasControls = false;
+
+  /** specify text to be user for the enable editing button's aria-label, defaults to `Click to edit` */
+  @Prop({ reflect: true }) intlEnableEditing = TEXT.intlEnablingEditing;
+
+  /** specify text to be user for the cancel editing button's aria-label, defaults to `Cancel` */
+  @Prop({ reflect: true }) intlCancelEditing = TEXT.intlCancelEditing;
+
+  /** specify text to be user for the confirm changes button's aria-label, defaults to `Save` */
+  @Prop({ reflect: true }) intlConfirmChanges = TEXT.intlConfirmChanges;
+
+  /** specify the scale of the inline-editable component, defaults to the scale of the wrapped calcite-input or the scale of the closest wrapping component with a set scale */
+  @Prop({ reflect: true }) scale?: "s" | "m" | "l";
+
+  /** specify the scale of the inline-editable component, defaults to the scale of the wrapped calcite-input or the theme of the closest wrapping component with a set theme */
+  @Prop({ reflect: true }) theme?: "light" | "dark";
+
+  //** when has-controls, specify a callback to be executed prior to disabling editing. when provided, loading state will be handled automatically. */
+  @Prop() onConfirmChanges: () => Promise<any>; // eslint-disable-line @stencil/decorators-style
+
+  //--------------------------------------------------------------------------
+  //
+  //  Getters/Setters
+  //
+  //--------------------------------------------------------------------------
+
+  get shouldShowControls(): boolean {
+    return this.editingEnabled && this.hasControls;
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  //--------------------------------------------------------------------------
+
+  componentWillLoad() {
+    this.inputElement = this.el.querySelector("calcite-input") as HTMLCalciteInputElement;
+    this.scale =
+      this.scale || this.inputElement.scale || getElementProp(this.el, "scale", undefined);
+    this.theme =
+      this.theme || this.inputElement.theme || getElementProp(this.el, "theme", undefined);
```

Comment:
☝️ Haven't used `getElementProp` much, but how is it different than just accessing `this.scale`?

Reply:
Needed to look up the tree, and this is what calcite input is doing as well. I "borrowed" the pattern from there.